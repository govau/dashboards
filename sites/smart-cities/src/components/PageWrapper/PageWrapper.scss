@import '~style/main';

.pageWrapper {
  @include for-desktop-up {
    padding-left: $sidebar-width;
  }
}

.main {
  min-height: 100vh; // stop the footer creeping up the page
}
