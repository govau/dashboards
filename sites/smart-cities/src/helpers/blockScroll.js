export default function blockScroll(shouldBlock) {
  if (shouldBlock) {
    document.body.classList.add('block-scroll');
  } else {
    document.body.classList.remove('block-scroll');
  }
}
