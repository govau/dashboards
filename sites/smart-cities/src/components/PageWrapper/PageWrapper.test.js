import React from 'react';
import { shallow } from 'enzyme';
import PageWrapper from './PageWrapper';
import blockScroll from '../../helpers/blockScroll';

jest.mock('../../helpers/blockScroll', () => jest.fn());
jest.mock('../FooterContainer/FooterContainer', () => 'FooterContainer');
jest.mock('../Header/Header', () => 'Header');
jest.mock('../SideBar/SideBar', () => 'SideBar');

const defaultProps = {
  categoryId: 'jobs',
  cityId: 'perth',
};

it('should match Snapshot', () => {
  const component = shallow(
    <PageWrapper {...defaultProps}>
      <div>render me</div>
    </PageWrapper>
  );

  expect(component.debug()).toMatchSnapshot();
});

describe('when the nav is opened', () => {
  it('should open the nav', () => {
    const component = shallow(
      <PageWrapper {...defaultProps}>
        <div>render me</div>
      </PageWrapper>
    );

    component.instance().setNavState(true); // pass true to open

    expect(component.state('navIsOpen')).toBe(true);
  });

  it('should block body scroll', () => {
    const component = shallow(
      <PageWrapper {...defaultProps}>
        <div>render me</div>
      </PageWrapper>
    );

    component.instance().setNavState(true); // pass true to open

    expect(blockScroll).toHaveBeenCalledTimes(1);
    expect(blockScroll).toHaveBeenCalledWith(true);
  });
});

describe('when the nav is closed', () => {
  it('should close the nav', () => {
    const component = shallow(
      <PageWrapper {...defaultProps}>
        <div>render me</div>
      </PageWrapper>
    );

    component.instance().setState({ isNavOpen: true });

    component.instance().setNavState(false); // pass false to close

    expect(component.state('navIsOpen')).toBe(false);
  });

  it('should unblock body scroll', () => {
    const component = shallow(
      <PageWrapper {...defaultProps}>
        <div>render me</div>
      </PageWrapper>
    );

    component.instance().setState({ isNavOpen: true });

    component.instance().setNavState(false); // pass false to close

    expect(blockScroll).toHaveBeenCalledTimes(1);
    expect(blockScroll).toHaveBeenCalledWith(false);
  });
});

it('should unblock scroll on unmount', () => {
  const component = shallow(
    <PageWrapper {...defaultProps}>
      <div>render me</div>
    </PageWrapper>
  );

  expect(blockScroll).toHaveBeenCalledTimes(0);

  component.unmount();

  expect(blockScroll).toHaveBeenCalledTimes(1);
  expect(blockScroll).toHaveBeenCalledWith(false);
});
