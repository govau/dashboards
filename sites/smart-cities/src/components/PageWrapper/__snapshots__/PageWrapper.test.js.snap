// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"pageWrapper\\">
  <div>
    <SideBar closeNav={[Function]} isOpen={false} cityId=\\"perth\\" />
  </div>
  <div>
    <Header openNav={[Function]} cityId=\\"perth\\" categoryId=\\"jobs\\" cityName={[undefined]} categoryColorName={[undefined]} onFeedbackPage={[undefined]} />
    <main className=\\"main\\">
      <div>
        render me
      </div>
    </main>
    <FooterContainer categoryId=\\"jobs\\" />
  </div>
</div>"
`;
